internal/backend/local/backend_local.go

```diff
@@ -400,9 +402,8 @@ func (b *Local) interactiveCollectVariables(ctx context.Context, existing map[st
 	// variable's value.
 	sort.Strings(needed) // prompt in lexical order
 	ret := make(map[string]backendrun.UnparsedVariableValue, len(vcs))
-	for k, v := range existing {
-		ret[k] = v
-	}
+	maps.Copy(ret, existing)
```

Comment:
these should all be `maps.Clone` when we're allocating the empty map just above

Reply:
Oh, that's a good point! Some of these were probably counting on never returning a nil map. Anything that feeds into `cty.ObjectVal` is an obvious candidate, which would be `map[string]cty.Value`, though I only see one of those in this diff. Maybe if you can easily prove the code is fine passing through a `nil` map you could use `Clone`, and leave a comment why it's `Copy` on any you're unsure about? I would just want at least a comment bc someone may eventually come through and make the same change again later.

Reply:
I added comments to the sections, where the map is returned, and thus may be used beyond the current scope. 

For the ones where the map is immediately written to in the current scope, I find it redundant to add the comment, and expecting that one would ensure to allocate a map before attempting to write to it.